R/fct_appdata.R

```diff
@@ -233,7 +233,17 @@ get_appdata <-  function(
     meta = metadata
 ){
   tableclasses <- gsub("create_table.", "", as.character(utils::methods("create_table")))
-  var_levels <- dplyr::distinct(meta$items_expanded, item_name, item_group) 
+  var_levels <- dplyr::distinct(meta$items_expanded, item_name, item_group)
+  form_types <- lapply(
+    c("common_forms", "study_forms", "general"),
+    \(x) {
+      lapply(metadata[x], \(y) {
+        cbind(form_type = x, item_group = unique(y$item_group))
+      }) |> 
+        do.call(what = rbind)
+    }) |> 
+    do.call(what = rbind) |> 
+    as.data.frame()
```

Comment:
I'm not opposed. I think if we come up with more utility for the field that could make sense. I simplified the creation of the object a little bit by using `Map()` instead of nested `lapply()`s. I think it looks a bit less awkward now.

Reply:
If you are not opposed to it, maybe it is better to add it to the metadata$items_expanded table. 

By doing so we don't even need to use `lapply` or `Map` in `get_appdata`.

In addition, I plan to remove the tabs `common_forms`, `general`, and `study_forms` from the metadata object in the future since it is redundant information (all this information is also in `items_expanded`)

Reply:
`form_type` column has been added to `items_expanded`